spec/unit/inputs/dead_letter_queue_spec.rb

```diff
@@ -23,6 +26,21 @@
     FileUtils.remove_entry_secure directory
   end
 
+  context "when clean_consumed is enabled" do
+    let(:config) { super().merge!("clean_consumed" => true) }
+
+    it "shouldn't register" do
+      expect {subject.register}.to_not raise_error
+    end
+
+    context "and commit_offsets is not" do
+      let(:config) { super().merge!("commit_offsets" => false) }
+      it "shouldn't register" do
```

Comment:
Nit: rspec [style guide](https://github.com/rubocop/rspec-style-guide#should-in-example-docstrings) recommends against using `should` or `shouldn't` in test descriptions

```suggestion
      it "raises a configuration error during register" do
```